SeQuant/core/space.hpp

```diff
@@ -464,19 +465,26 @@ class IndexSpace {
 inline const IndexSpace IndexSpace::null;
 inline const IndexSpace::Attr IndexSpace::Attr::null;
 
-/// @return true if type2 is included in type1, i.e. intersection(type1, type2)
-/// == type2
+inline auto hash_value(const IndexSpace &space) {
+  auto result = hash_value(static_cast<std::int64_t>(space.attr()));
+  hash::combine(result, space.base_key());
+  hash::combine(result, space.approximate_size());
```

Comment:
I don't think we'll want the size to be part of the hash. Imo, having two index spaces that differ only in their size doesn't make any sense :thinking: 

And I think the same argument kinda holds for the base label as well

This is particularly true as these properties are not taken into account in `IndexSpace::operator==` and I believe we should definitely match the hashing with identity checking